data-upgrade-users/src/main/java/org/exoplatform/migration/UserSetExternalInGateinPortal.java

```diff
@@ -63,9 +72,14 @@ public void processUpgrade(String oldVersion, String newVersion) {
             if (profile==null) {
               profile=organizationService.getUserProfileHandler().createUserProfileInstance(username);
             }
-            profile.setAttribute(UserProfile.OTHER_KEYS[2],"true");
-            organizationService.getUserProfileHandler().saveUserProfile(profile,true);
-            LOG.debug("External info added in gatein profile for user {}, {}ms", username, System.currentTimeMillis() - startTimeForUser);
+            // do not update the profiles of users who already have the correct external property value
+            if (profile.getAttribute(UserProfile.OTHER_KEYS[2]) != null && String.valueOf(true).equals(profile.getAttribute(UserProfile.OTHER_KEYS[2]))) {
+              LOG.debug("External info already set in gatein profile for user {}", username);
+            } else {
+              profile.setAttribute(UserProfile.OTHER_KEYS[2],"true");
+              organizationService.getUserProfileHandler().saveUserProfile(profile,true);
+              LOG.debug("External info added in gatein profile for user {}, {}ms", username, System.currentTimeMillis() - startTimeForUser);
+            }
```

Comment:
please add log here to show progression: nb of users executed/ total number of users

Reply:
done!